### ISSUE TITLE
ResizeObserver not working with multiple images and tailwindcss 

### ISSUE THREAD

**Issue by lakkes-ra** [spatie/statamic-responsive-images#90]
## Problem
The first image gets blurred / not loaded properly, when there're multiple (responsive:)images on the page. I get the following error:
```js
Uncaught TypeError: Cannot read properties of undefined (reading 'observe')
    at HTMLImageElement.onload

```
After a while we found out that we get rid of that issue when we do not load the tailwind stylesheet.

## Workaround
To solve the problem temporarily, we downgraded to v2.5.5, because the use of the ResizeObserver came with version 2.6

## Environment
`php 8.0` `node 15.14` `statamic 3.2.8`

## Reproduction

```html
<!doctype html>
<html>
  
  <head>
      <link href="https://unpkg.com/tailwindcss@^2/dist/tailwind.min.css" rel="stylesheet">
  </head>
  
  <body>
      <main>
          {{ responsive:image_01 }} <!-- this one get's blurry -->
          {{ responsive:image_02 }} 
      </main>
  </body>

</html>
```

**Comment by riasvdv** [comment]
Does this happen in every browser?

**Comment by lakkes-ra** [comment]
We tested it in Chrome, Firefox, Edge and Safari. All have the same issue.

**Comment by mikemartin** [comment]
Just experienced this too. Been using this responsive images addon a long time and have never hit this one:

```
 <div class="grid md:grid-cols-12 items-start lg:items-center">
        <div class="md:col-span-6 flex flex-col">
          {{ responsive:hero_image glide:width="1200" class="max-w-sm md:max-w-none w-full" alt="{ hero_image:alt }" }}
        </div>
        <div class="mt-12 md:col-start-7 md:col-span-5">
          ...
        </div>
      </div>
```

<img width="532" alt="Screen Shot 2021-12-08 at 2 47 52 pm" src="https://user-images.githubusercontent.com/414211/145144952-b24fa417-7c1b-48ec-9ebd-f1bde072ebae.png">




```
Statamic 3.2.25 Pro
Laravel 8.75.0
PHP 7.4.25
aryehraber/statamic-captcha 1.7.0
doublethreedigital/duplicator 2.0.1
edalzell/forma 1.0
mikemartin/helpscout-beacon 1.0.2
rias/statamic-redirect 1.9.0
silentz/akismet 3.0.8
spatie/statamic-responsive-images 2.8.5
webographen/statamic-widget-continue-editing 1.0.0
```

**Comment by taylorcammack** [comment]
Any update on this? I'm having the same issue with a current project running `2.8.5`, and have had it pop up on about 1 out of 4 projects or so. Using an older version works, but I'd prefer not to have to continue to go that route 😬

**Comment by riasvdv** [comment]
I've lost track of this one, I'll try to get a look at it next week, PRs are also always welcome if you'd find a solution in the meantime.

**Comment by bdbch** [comment]
Downgrading the package to `2.5.5` didn't help me. What I did now is take this:

```
window.responsiveResizeObserver = new ResizeObserver((entries) => {
    entries.forEach(entry => {
        const imgWidth = entry.target.getBoundingClientRect().width;
        entry.target.parentNode.querySelectorAll('source').forEach((source) => {
            source.sizes = Math.ceil(imgWidth / window.innerWidth * 100) + 'vw';
        });
    });
});

const responsiveImages = document.querySelectorAll('picture img[onload]');
responsiveImages.forEach(responsiveImage => {
    responsiveResizeObserver.onload = null;
    responsiveResizeObserver.observe(responsiveImage);
})
```

and put it into my own javascript. In general I think the image elements should not have their observer binding via the onload tag. it seems like there are potential problems with how browsers read the DOM?

In Chrome I almost always had the issue that images are not loading. Edge had them sometimes and Firefox almost always showed them correctly.

https://github.com/spatie/statamic-responsive-images/blob/main/resources/views/responsiveImage.blade.php#L3

I think here we should rather do what I did - also binding the onload in the script tag OR create a tag like `{{ responsiveScripts }}` which users can then put into their `<head>` or whereever they want so we don't have to rely on that @once directive.

What are your thoughts?

**Comment by riasvdv** [comment]
I've changed the ResizeObserver a bit as suggested by @bdbch, I still cannot reproduce this issue, so could you guys check if this solves it?

If not I'll think of a different solution.

**Comment by bdbch** [comment]
Thanks! I'll give it a try as soon as I find time. It is a weird issue yeah - on another website I didn't run into this issue but with the current site I'm working on (large onepage landingpage) it seems to not work in Chrome for most of the time.

**Comment by mkriegeskorte** [comment]
Had the same issue as @bdbch described. The fix worked well for me!

**Comment by lakkes-ra** [comment]
My first tests seem to have fixed the problem as well. After the update the responsive image height and width somehow broke, which isn't the case if I use v2.8.5. It tells me »Uncaught TypeError: window.responsiveResizeObserver is undefined«.

**Comment by riasvdv** [comment]
The height issue should be fixed in 2.9.1, sorry about that 😬
